contributors/design-proposals/statefulset-update.md

```diff
@@ -0,0 +1,821 @@
+and it will not rule out the future implementation of the out of scope use 
+cases.
+
+### In Scope
+- As the administrator of a stateful application, in order to vertically scale 
+my application, I want to update resource limits or requested resources.
+- As the administrator of a stateful application, in order to deploy critical 
+security updates, break fix patches, and feature releases, I want to update 
+container images.
+- As the administrator of a stateful application, in order to update my 
+application's configuration, I want to update environment variables, container 
+entry point commands or parameters, or configuration files.
+- As the administrator of the logging and monitoring infrastructure for my 
+organization, in order to add logging and monitoring side cars, I want to patch
+containers to add images.
+
+### Out of Scope
+- As the administrator of a stateful application, in order to increase the 
+applications storage capacity, I want to update PersistentVolumes.
+- As the administrator of a stateful application, in order to update the 
+network configuration of the application, I want to update Services and 
+container ports in a consistent way.
+- As the administrator of a stateful application, when I scale my application 
+horizontally, I want associated PodDisruptionBudgets to be adjusted to 
+compensate for the application's scaling.
+
+## Assumptions
+ - StatefulSet update must support singleton StatefulSets. However, an update in
+ this case will cause a temporary outage. This is acceptable as a single 
+ process application is, by definition, not highly available.
+ - Disruption in Kubernetes is controlled by PodDisruptionBudgets. As 
+ StatefulSet updates progress one Pod at a time, and only occur when all 
+ other Pods have a Status of Running and a Ready Condition, they can not 
+ violate reasonable PodDisruptionBudgets.
+ - Without priority and preemption, there is no guarantee that an update will 
+ not block due to a loss of capacity or due to the scheduling of another Pod
+ between Pod termination and Pod creation. This is mitigated by blocking the 
+ update when a Pod fails to schedule. Remediation will require operator 
+ intervention. This implementation is no worse than the current behavior with 
+ respect to eviction.
+ - We will eventually implement a signal that is delivered to Pods to indicate 
+ the 
+ [reason for termination](https://github.com/kubernetes/community/pull/541).
+ - StatefulSet updates will use the methodology outlined in the 
+ [controller history](https://github.com/kubernetes/community/pull/594) proposal 
+ for version tracking, update detection, and rollback detection.
+ This will be a general implementation, usable for any Pod in a Kubernetes 
+ cluster. It is, therefore, out of scope to design such a mechanism here.
+ - Kubelet does not support resizing a container's resources without terminating 
+ the Pod. In place resource reallocation is out of scope for this design. 
```

Comment:
Curious if there is a proposal for inplace reallocation. Not that i need this, just curious, how would one do this ?